glean-core/src/error.rs

```diff
@@ -0,0 +1,109 @@
+use std::fmt::{self, Display};
+use std::io;
+use std::result;
+
+use failure::{self, Backtrace, Context, Fail};
+
+use ffi_support::ExternError;
+
+use rkv::error::StoreError;
+
+/// A specialized [`Result`] type for this crate's operations.
+///
+/// This is generally used to avoid writing out [Error] directly and
+/// is otherwise a direct mapping to [`Result`].
+///
+/// [`Result`]: https://doc.rust-lang.org/nightly/std/result/enum.Result.html
+/// [`Error`]: std.struct.Error.html
+pub type Result<T> = result::Result<T, Error>;
+
+/// A list enumerating the categories of errors in this crate.
+///
+/// This list is intended to grow over time and it is not recommended to
+/// exhaustively match against it.
+///
+/// It is used with the [`Error`] struct.
+///
+/// [`Error`]: std.struct.Error.html
+#[derive(Debug, Fail)]
+pub enum ErrorKind {
+    /// Any error not part of this list.
+    #[fail(display = "Generic error.")]
+    Other,
```

Comment:
Does this associate a message with the "Other" errors?  I think we should probably require a message for any misc. error something might return.

Reply:
Actually, we currently don't use the `Other` variant at all. We only handle rkv and io errors.
I'll remove that variant for now.

Reply:
Makes sense.  We can always add something back in if we realize we need our own error type.